dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/RequestItemRepository.java

```diff
@@ -0,0 +1,146 @@
+        throw new RepositoryMethodNotImplementedException(RequestItemRest.NAME, "findAll");
+    }
+
+    @Override
+    @PreAuthorize("permitAll()")
+    public RequestItemRest createAndReturn(Context ctx) {
+        // Fill a RequestItemRest from the client's HTTP request.
+        HttpServletRequest req = getRequestService()
+                .getCurrentRequest()
+                .getHttpServletRequest();
+        ObjectMapper mapper = new ObjectMapper();
+        RequestItemRest rir;
+        try {
+            rir = mapper.readValue(req.getInputStream(), RequestItemRest.class);
+        } catch (IOException ex) {
+            throw new UnprocessableEntityException("error parsing the body", ex);
+        }
+
+        // Create the item request model object from the REST object.
+        String token;
+        try {
+            String bitstreamId = rir.getBitstreamId();
+            if (isBlank(bitstreamId)) {
+                throw new IncompleteItemRequestException("A bitstream ID is required");
+            }
+            Bitstream bitstream = bitstreamService.find(ctx, UUID.fromString(bitstreamId));
+            if (null == bitstream) {
+                throw new IncompleteItemRequestException("That bitstream does not exist");
+            }
+
+            String itemId = rir.getItemId();
+            if (isBlank(itemId)) {
+                throw new IncompleteItemRequestException("An item ID is required");
+            }
+            Item item = itemService.find(ctx, UUID.fromString(itemId));
+            if (null == item) {
+                throw new IncompleteItemRequestException("That item does not exist");
+            }
+
+            boolean allFiles = rir.isAllfiles();
+
+            String email = rir.getRequestEmail();
+            if (isBlank(email)) {
+                throw new IncompleteItemRequestException("A submitter's email address is required");
+            }
+
+            String username = rir.getRequestName();
+            String message = rir.getRequestMessage();
+
+            token = requestItemService.createRequest(ctx, bitstream, item,
+                    allFiles, email, username, message);
+        } catch (SQLException ex) {
+            throw new RuntimeException("Item request not created.", ex);
+        }
+
+
+        // Some fields are given values during creation, so return created request.
+        RequestItem ri = requestItemService.findByToken(ctx, token);
+        ri.setAccept_request(false); // Not accepted yet.  Must set:  DS-4032
+        requestItemService.update(ctx, ri);
+        return requestItemConverter.convert(ri, new DefaultProjection());
+    }
+
+    // NOTICE:  there is no service method for this -- requests are never deleted?
+    @Override
+    public void delete(Context context, String token)
+            throws AuthorizeException, RepositoryMethodNotImplementedException {
+        throw new RepositoryMethodNotImplementedException(RequestItemRest.NAME, "delete");
```

Comment:
It looks like you've implemented a delete service method in `RequestItemService.delete()`.  Should we therefore update this REST endpoint to support deletion?  That'd also seemingly allow us to close #7417 

If there's a reason you'd rather not implement deletion in this PR, that's OK.  Was just noting the discrepancy between this method and the new `RequestItemService.delete()` method.

Reply:
This would allow anyone to delete any request, if he can guess a token.  Item requests bypass the usual DSpace access controls -- that's what they are for.  I doubt that requests would ever be deleted via REST calls.

We need some discussion of how, and whether, requests are deleted.  I would suggest discussion in #7417.

The `RequestItemService.delete()` method was needed for testing.